protecode/scanning.py

```diff
@@ -464,53 +490,108 @@ def result_matches(resource: cm.Resource, result: pm.AnalysisResult):
 def _retrieve_existing_scan_results(
     protecode_client: protecode.client.ProtecodeApi,
     group_id: int,
-    resources: tuple[cnudie.iter.ResourceNode],
+    resource_groups: typing.Iterable[tuple[cnudie.iter.ResourceNode]],
 ) -> dict[
-    tuple[cm.ComponentIdentity, cm.ResourceIdentity, cm.ArtefactType|str],
+    tuple[str, cm.ResourceIdentity, cm.ArtefactType|str],
     list[pm.Product],
 ]:
     # This function populates a dict that contains all relevant scans for all artifacts in a given
     # protecode group.
     # The created dict is later used to lookup existing scans when creating scan requests
     scan_results = dict()
-    for resource in resources:
-        c = resource.component
-        r = resource.resource
-
-        query_data = protecode.util.component_artifact_metadata(
-            component=c,
-            artefact=r,
-            omit_component_version=False,
-            omit_resource_version=False,
-        )
+    for resource_group in resource_groups:
+        # groups are assumed to belong to same component + resource version name, and type so it
+        # is okay to choose first one as representative
+        c = resource_group[0].component
+        r = resource_group[0].resource
+
+        # special-handling for vm-image-rootfs - we keep different component-versions in parallel
+        # for those
+        if (
+            isinstance(r.type, str)
+            and r.type.startswith('application/tar+vm-image-rootfs')
```

Comment:
ditto (also: in previous occurrence, str-equality is checked; in this place, we check for startswith; -> should  be consistent)